### PR TITLE
Topic: CommandLine

### PR DESCRIPTION
**Pull request by kkoppolu1**
Add command line arguments for structs

### CODE REVIEW

**Comment by kkoppolu1** [comment]
Reviews in this chain:
└**https://github.com/COVESA/vss-tools/pull/220** **Topic: CommandLine**

**Comment by kkoppolu1** [comment]
| # | head | base | diff | date | summary |
| - | - | - | - | - | - |
| 0 | [445078fc](/kkoppolu1/vss-tools/commit/445078fc0abdfa65f1ff09a1357f7d67b5de8443) | [7d095715](/kkoppolu1/vss-tools/commit/7d09571557111d0220e77976dc8d975f3fcfc53d) | [diff](/kkoppolu1/vss-tools/compare/7d09571557111d0220e77976dc8d975f3fcfc53d..445078fc0abdfa65f1ff09a1357f7d67b5de8443) | Feb 3 23:06 PM |  5 files changed, 128 insertions(+), 43 deletions(-) |

File: vspec/loggingconfig.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python3
+
+# (c) 2022 BMW Group
```
Comment (erikbosch) [inline vspec/loggingconfig.py:3]: As this is a totally new file, not copied from any existing one you could if you like remove BMW/BoSch/JLR here. If you like you could instead add your own organization.

(The practical approach so far concerning copyright have been that if you do significant contributions to a file then you could add your organization to copyright, if you like)

**Review by erikbosch — APPROVED** [review]
Looking good. I like the refactoring to use logging instead of simple prints

**Comment by kkoppolu1** [comment]
## Incremental Change for struct support: Add command line arguments

### Description
Add command-line arguments to the VSS parser for struct support.
Details of the proposal in https://github.com/COVESA/vehicle_signal_specification/pull/511

Two new arguments are added:

1. Input file containing the types
2. Output file for the types in the requested format (JSON for example)

The following checks are added:

1. Overlays cannot be specified when data type support arguments are provided
2. Only JSON format can be specified when data type support arguments are provided
3. When the feature is requested, the user is informed that it is experimental.
4. After all checks have passed, a `NotImplemented` exception is thrown.

### Miscellaneous
- Run autopep8 on all touched files
- Replace print statements with logging in vspec2x.py and vspec2json.py

Example output:
```
INFO     Output to json format
INFO     Known extended attributes: 
WARNING  From VSS 4.0 the default behavior for printing uuid will change. Consider using --uuid or --no-uuid.
WARNING  Use of default VSS unit file is deprecated, please specify the unit file you want to use with the -u argument!
INFO     Loading vspec from /home/kkoppolu/workplace/vehicle_signal_specification/spec/VehicleSignalSpecification.vspec...
INFO     Calling exporter...
INFO     Generating JSON output...
INFO     Serializing compact JSON...
INFO     All done.
```

### Testing
- Ran pytest
- Ran JSON converter using the standard catalog.


Signed-off-by: Krishna Koppolu <[krishnachaitanya.85@gmail.com](mailto:krishnachaitanya.85@gmail.com)>

File: vspec/loggingconfig.py [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+# Convert vspec files to various other formats
+#
+
+import logging
+
+def initLogging():
+    """
+    Initialize logging
+    """
+    logging.basicConfig(level=logging.DEBUG, format='%(levelname)-8s %(message)s')
```
Comment (erikbosch) [inline vspec/loggingconfig.py:16]: For now I assume DEBUG is OK as default level. But do we want to keep  it as default in the long run, or should rather INFO be the default level?

Comment (kkoppolu1) [reply vspec/loggingconfig.py:16]: Yes. a command line arg is pretty common.

**Pull request merged**